jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1158,13 +1199,25 @@ public Method getDeclaredMethod(String name, Class<?>... parameterTypes) throws
 /**
  * Answers the class which declared the class represented
  * by the receiver. This will return null if the receiver
- * is a member of another class.
+ * is not a member of another class.
  *
  * @return		the declaring class of the receiver.
  */
 @CallerSensitive
 public Class<?> getDeclaringClass() {
-	Class<?> declaringClass = getDeclaringClassImpl();
+	if (cachedDeclaringClassOffset == -1) {
+		cachedDeclaringClassOffset = getFieldOffset("cachedDeclaringClass"); //$NON-NLS-1$
+	}
+	if (cachedDeclaringClass == null) {
+		Class<?> localDeclaringClass = getDeclaringClassImpl();
+		writeFieldValue(cachedDeclaringClassOffset, localDeclaringClass == null ? Dummy.class : localDeclaringClass);
```

Comment:
While concise, this pattern is harder to set a breakpoint on.
Can this be written as:
```
if (localDeclaringClass == null) {
  localDeclaringClass = Dummy.class;
}
writeFieldValue(cachedDeclaringClassOffset, localDeclaringClass);
```
That's much easier to breakpoint on the case where we're writing null